packages/providers/src/utils/test-utils/launchTestNode.ts

```diff
@@ -0,0 +1,128 @@
+  cleanup: () => Promise<void>;
+  ip: string;
+  port: string;
+}>;
+
+/**
+ * Launches a fuel-core node locally (`127.0.0.1`)
+ * @param consensusKey - the consensus key to use.
+ * @param port - the port to bind to. (optional, by default the OS assigns it)
+ * @param args - additional arguments to pass to fuel-core.
+ * @param useSystemFuelCore - whether to use the system fuel-core binary or the one provided by the \@fuel-ts/fuel-core package.
+ * */
+export const launchTestNode = async ({
+  consensusKey = '0xa449b1ffee0e2205fa924c6740cc48b3b473aa28587df6dab12abc245d1f5298',
+  port = '0',
+  args = defaultFuelCoreArgs,
+  useSystemFuelCore = false,
+  chainConfig = defaultChainConfig,
+  logger,
+}: Partial<LaunchTestNodeOptions> = {}): LaunchNodeResult =>
+  // eslint-disable-next-line no-async-promise-executor
+  new Promise(async (resolve, reject) => {
+    const command = useSystemFuelCore ? 'fuel-core' : './node_modules/.bin/fuels-core';
+
+    const tempDirPath = path.join(os.tmpdir(), '.fuels-ts', randomUUID());
+
+    if (!fsSync.existsSync(tempDirPath)) {
+      fsSync.mkdirSync(tempDirPath, { recursive: true });
+    }
+    const chainConfigPath = path.join(tempDirPath, '.chainConfig.json');
+
+    // Write a temporary chain configuration file.
+    await fs.writeFile(chainConfigPath, JSON.stringify(chainConfig), 'utf8');
+
+    const child = spawn(command, [
+      'run',
+      '--db-type',
+      'in-memory',
+      '--consensus-key',
+      consensusKey,
+      '--chain',
+      chainConfigPath as string,
+      '--ip',
+      '127.0.0.1',
+      '--port',
+      port,
+      ...args,
+    ]);
+
+    function removeSideffects() {
+      child.stdout.removeAllListeners();
+      child.stderr.removeAllListeners();
+      spawnSync('rm', ['-rf', tempDirPath]);
+    }
+
+    // Cleanup function where fuel-core is stopped.
+    const cleanup = () =>
+      new Promise<void>((resolveFn, rejectFn) => {
+        kill(Number(child.pid), (err) => {
+          removeSideffects();
+
+          if (err) rejectFn(err);
+          resolveFn();
+        });
+      });
+
+    // Process exit.
+    process.on('exit', cleanup);
+
+    // Catches ctrl+c event.
+    process.on('SIGINT', cleanup);
+
+    // Catches "kill pid" (for example: nodemon restart).
+    process.on('SIGUSR1', cleanup);
+    process.on('SIGUSR2', cleanup);
+
+    // Catches uncaught exceptions.
+    process.on('uncaughtException', cleanup);
+
+    child.stderr.setEncoding('utf8');
+
+    // This string is logged by the client when the node has successfully started. We use it to know when to resolve.
+    const graphQLStartSubstring = 'Binding GraphQL provider to';
+
+    // Look for a specific graphql start point in the output.
+    child!.stderr.on('data', (chunk: string) => {
+      if (logger) logger(chunk);
+
+      if (chunk.indexOf(graphQLStartSubstring) !== -1) {
+        const [nodeIp, nodePort] = chunk.split(' ').at(-1)!.trim().split(':');
```

Comment:
The fuel-core node logs on which url it's hosted, which enabled me to use port nr. `0` as the default, which gives rein to the OS to assign any port it wants to the node. This way we don't need `getPortPromise`.